### PR TITLE
Make FXAlbum.coverProperty trigger lazy cover load on observe

### PR DESCRIPTION
**Pull request by octaviospain**
## Summary

Binding an album-grid cell to `FXAlbum.coverProperty` (via `addListener`) previously never triggered the lazy cover load — the cover only appeared if the consumer explicitly called `getCoverImageBytes()`. This aligns album-cover behavior with the proven track-cover behavior in `FXAudioItem.coverImageProperty`, so observing the property itself drives the load.

`FXAlbum.coverProperty` is now an anonymous `SimpleObjectProperty<Optional<Image>>` subclass that overrides `addListener` (both overloads) and `get()` to invoke the existing `coverImageBytes` resolution path on first observation, behind a one-shot `@Transient @Volatile` latch. The latch is set **before** reading `coverImageBytes` so the inline `coverProperty.set(...)` in the headless `runOnFxThread` fallback cannot re-enter the overridden `get()`/`addListener` and re-trigger resolution.

Cover resolution stays a side effect: the first `get()` returns the current (empty) `Optional`, and bound listeners receive the resolved `Image` on the next JavaFX pulse via the unchanged `publishCoverImage` → `runOnFxThread { coverProperty.set(...) }` path — identical to `FXAudioItem.coverImageProperty`.

## Changes

- **`FXAlbum.kt`** — `coverProperty` converted to a one-shot-latched triggering `SimpleObjectProperty<Optional<Image>>` subclass. The `coverImageBytes` getter, `coverBytesRef`, `noCover`, `publishCoverImage`, `runOnFxThread`, and `firstCoverImageBytes` resolution path are unchanged. No orphan-retry / metadata-guard branch is added — `FXAlbum`'s track list is final at construction and always resolvable immediately.
- **`ObservableAlbum.kt`** — `coverProperty` KDoc rewritten to document the observe-triggers-load contract, retaining the no-cover-stays-empty and JavaFX-Application-Thread guarantees.
- **`FXAlbumTest.kt`** — cover tests fixed and extended: fire-once-on-observe for both `addListener` and `get()`, and a no-cover album stays empty without re-probing under repeated observation.

## Verification

- [x] `gradle :music-commons-fx:compileKotlin compileTestKotlin ktlintCheck` — passes
- [x] `gradle :music-commons-fx:test` — full FX module suite passes (including the extended `FXAlbumTest`)

No new public types or API surface; no change to `coverImageBytes` resolution logic.

Closes #163

### CODE REVIEW

File: music-commons-fx/src/main/kotlin/net/transgressoft/commons/fx/music/audio/FXAlbum.kt [hunk 1]
```diff
@@ -118,8 +120,25 @@ internal class FXAlbum(
...
     override val coverProperty: ReadOnlyObjectProperty<Optional<Image>>
-        field = SimpleObjectProperty(this, "cover", Optional.empty())
+        field = LazyObservationObjectProperty(this@FXAlbum, "cover", Optional.empty<Image>(), ::triggerLazyCoverLoad)
+
+    private fun triggerLazyCoverLoad() {
+        if (coverObservationTriggered) return
+        coverObservationTriggered = true
+        // Resolve off the JavaFX thread: coverImageBytes reads the first track's art and decodes the
+        // image on the cover-load worker, publishing coverProperty.set(...) back on the JavaFX thread.
+        CoverLoadExecutor.execute { coverImageBytes }
```
[the remark this answers is not in the document]
Comment (octaviospain) [reply music-commons-fx/src/main/kotlin/net/transgressoft/commons/fx/music/audio/FXAlbum.kt:141]: Done — switched the latch to `AtomicBoolean` and claim it with `compareAndSet(false, true)` in `triggerLazyCoverLoad()`, so concurrent first observations can no longer both enqueue a cover load. Applied the same fix to `FXAudioItem.triggerLazyCoverLoad()` (preserving its orphan-retry reset via `set(coverLoaded)` in the finally block).

File: music-commons-fx/src/test/kotlin/net/transgressoft/commons/fx/music/audio/FXAlbumTest.kt [hunk 2]
```diff
@@ -222,15 +222,88 @@ internal class FXAlbumTest : StringSpec({
...
+        eventually(2.seconds) {
+            WaitForAsyncUtils.waitForFxEvents()
+            fxAlbum.coverProperty.get() shouldBePresent { }
+            changeCount shouldBe 1
+        }
+
+        // A second listener attach must not re-trigger resolution; changeCount stays at 1.
+        fxAlbum.coverProperty.addListener { _, _, _ -> }
         WaitForAsyncUtils.waitForFxEvents()
+        changeCount shouldBe 1
```
[the remark this answers is not in the document]
Comment (octaviospain) [reply music-commons-fx/src/test/kotlin/net/transgressoft/commons/fx/music/audio/FXAlbumTest.kt:275]: Done — added a dispatch-level seam: `CoverLoadExecutor` now tracks submitted-task count (`submittedTaskCount` / `resetSubmittedTaskCount()`). The three cover tests reset the counter before observing the album and assert `submittedTaskCount == 1`, so a duplicate submission that resolved to the same bytes would now fail the test rather than slip through a final-state check. The no-cover test asserts the single probe is dispatched once and then latched (no re-probe).

**Comment by octaviospain** [comment]
Addressed the outside-diff comment on `music-commons-fx/.../FXAudioItem.kt` (349-382, "make the observation latch atomic"): `coverObservationTriggered` is now an `AtomicBoolean` claimed via `compareAndSet(false, true)`, so concurrent observations cannot both enqueue a cover load. The orphan-retry behavior is preserved — the `metadataIO == null` early-return still does not consume the latch, and the `finally` block resets it with `set(coverLoaded)` so a coverless-yet-unwired item retries after the back-ref is attached while a loaded/known-coverless item stays settled.
